### PR TITLE
Dagger on Kubernetes

### PR DESCRIPTION
**Pull request by jlongtine**
Signed-off-by: Joel Longtine <joel@dagger.io>

### CODE REVIEW

**Comment by gerhard** [comment]
This is shaping up nicely! I know that this is still in draft, sharing what I imagine we will end up with (it's OK to skip the parts which don't resonate)
- a sub-section that makes it obvious this is AWS EKS specific
  - we should encourage contributions for other providers, e.g. GKE, Talos on Equinix Metal (I would a guy that wants to add this one 😉), etc.
- a high-level overview in no more than 500 words which will include:
  - pros & cons of this approach
- briefly mention Dagger with Dagger Cloud approach
  - I am keen to add a placeholder on https://docs.dagger.cloud for this (we can add the content later - the diagram is sufficient)

It would be awesome to get this merged tomorrow 💪

**Comment by gerhard** [comment]
WDYT @jlongtine, will this PR close this issue?
- https://github.com/dagger/dagger/issues/3140

**Comment by shykes** [comment]
Should I wait for some text to be added to the guide before reviewing it?

**Comment by d3rp3tt3** [comment]
@jlongtine and @gerhard , let's hold on merging this until we talk about how this fits into our overall guidance for running Dagger in production and how the content works into docs.dagger.io.

**Comment by jlongtine** [comment]
> Should I wait for some text to be added to the guide before reviewing it?

@shykes Yeah, it's not quite ready for review. Just wanting to keep people looped in as I work on the guide.

**Comment by jlongtine** [comment]
> WDYT @jlongtine, will this PR close this issue?
> 
> * [🖐️ How to run dagger inside a container ?  #3140](https://github.com/dagger/dagger/issues/3140)

@gerhard I think this will help with most of those requests... but not any of the interest in rootless mode.

**Comment by MChorfa** [comment]
This an amazing path forward to help  ML engineers/DS  to iterate quickly on the ML Model lifecycle :) 
Can you imagine what would unlock with GPU support?

**Comment by gerhard** [comment]
> @MChorfa: Can you imagine what would unlock with GPU support?

Yes! cc @matiasinsaurralde
- https://github.com/dagger/dagger/issues/4675

File: docs/current/guides/281969-dagger-on-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+- [**GitHub Actions Runner Controller**](https://github.com/actions/actions-runner-controller): A Kubernetes controller that manages deploying and scaling GitHub Actions runners as Kubernetes pods.
+- **[Karpenter](https://karpenter.sh/)**: Used for auto-scaling our Kubernetes nodes.
+- **Dagger**
+
+## Process
+<!-- TODO: Add costs for running Dagger on a cluster? -->
+
+1. **Set up a Kubernetes Cluster.** Given the number of ways that an cluster can be provisioned, we will not cover this in detail. If a quick start is needed, we recommend checking out [`eksctl`](https://eksctl.io/) for quickly provisioning an AWS EKS cluster.
+
+1. **Set up runners for your CI tool in your Kubernetes cluster**: Runners are the workhorses of your CI/CD system. They execute the jobs that are defined in your pipeline. Depending on your CI tool, the process of setting up runners may vary. For GitHub, you can follow the instructions in the [GitHub Actions documentation](https://github.com/actions/actions-runner-controller#getting-started).
```
Comment (vikram-dagger) [inline docs/current/guides/281969-dagger-on-kubernetes.md:91]: Could we add similar links for Gitlab and Drone?

File: docs/current/guides/281969-dagger-on-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+We're using a DaemonSet for the Dagger Engine to
+- best utilize the NVMe hard drives of our worker nodes
+- reduce the amount of network latency and bandwidth requirements
+- simplify routing of Dagger SDK and CLI requests
+
+We use the `_EXPERIMENTAL_DAGGER_RUNNER_HOST` environment variable to point to the Dagger Engine DaemonSet socket that we've mounted into our GitHub Runners. This ensures that our runners will use the local Dagger Engine for pipeline execution.
+
+## Design Decisions
+
+- We've chosen to use a Dagger Cloud Cache, which allows us to push cached layers up to the cloud and retrieve them at high bandwidth.
```
Comment (vikram-dagger) [inline docs/current/guides/281969-dagger-on-kubernetes.md:204]: What if the user doesn't have Dagger Cloud? Is there an alternative caching mechanism?

File: docs/current/guides/281969-dagger-on-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+              - key: github-actions-runner
+                operator: Exists
+      securityContext:
+        runAsUser: 0
+        runAsGroup: 1001
+        fsGroup: 1001
+        fsGroupChangePolicy: "OnRootMismatch"
+      initContainers:
+        - name: download-bin
+          image: {{ .Values.dagger.engine.image.repository }}:{{ .Values.dagger.engine.image.tag }}
```
Comment (nfisher) [inline docs/current/guides/281969-dagger-on-kubernetes.md:140]: You might either want to link to a Helm chart or provide a static file. IME it seems to be an even spread between folks using Kustomize and Helm.

Comment (d3rp3tt3) [reply docs/current/guides/281969-dagger-on-kubernetes.md:140]: There's a PR in progress for a Helm chart here too: #5505

**Comment by d3rp3tt3** [comment]
Joel and Vikram are still working on docs for running Dagger with Kubernetes. I know it might be confusing about why the PR is closed. The updated docs will be more comprehensive. A new PR will be coming soon.

**Pull request closed without merging**